src/popsift/s_orientation.cu

```diff
@@ -206,6 +207,7 @@ void ori_par( const int           octave,
 
     int2 best_index = make_int2( threadIdx.x, threadIdx.x + 32 );
 
+    __syncthreads();
```

Comment:
The syncthreads should be as early as possible, not just-in-time, to allow also old cards to do make_int2 in lockstep. Better move to line 207 right after the for loop.

I thought that I had it covered because all threads enter this loop:
for( int bin = threadIdx.x; popsift::any( bin < ORI_NBINS ); bin += blockDim.x )  {
... }
and inside they use predicates instead of ifs. I suppose that post-RTX cards do this differently.

Reply:
Moved up to L207 as suggested in d0edb05.